x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/InferenceResultsProvider.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.queries;
+
+import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.inference.InferenceResults;
+
+import java.util.Collection;
+
+public interface InferenceResultsProvider extends NamedWriteable {
```

Comment:
Nitpick: Could we please add a class comment here?

Reply:
Is it really needed, can we use a plain map instead of adding another named writeable? If we have a single inference id then Map.of() is enough too.

Reply:
++ making this a plain map instead of introducing `InferenceResultsProvider ` and `MapInferenceResultProvider` makes sense to me!